aws/resource_aws_lambda_event_source_mapping.go

```diff
@@ -170,7 +175,19 @@ func resourceAwsLambdaEventSourceMappingDelete(d *schema.ResourceData, meta inte
 		UUID: aws.String(d.Id()),
 	}
 
-	_, err := conn.DeleteEventSourceMapping(params)
+	err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err := conn.DeleteEventSourceMapping(params)
+		if err != nil {
+			if awserr, ok := err.(awserr.Error); ok {
```

Comment:
Minor nitpick: we have a helper function available for this type of logic and also prefer to use the SDK available constants, e.g. `isAWSErr(err, lambda.ErrCodeResourceInUseException, "")`

Reply:
Fixed.